### PR TITLE
Black settings and CI 

### LINKED ISSUE #78: Formatting all files in systems using Black
Depends on the #77  issue and waiting for PR #80 in order to be merged (although not really necessary but it's generated based on the use of the pytest-black automate file formatting so It would be wiser to wait until it will be merged to check whether the test passed or not).

### LINKED ISSUE #79: Adding black to the workflow in GitHub - CI
writing a .yml file to implement an automatic file as we did with flake8 to run tests on the code we present in each PR.

### PR DESCRIPTION
**Pull request by maayanhd**
- Adding black formatting check to the workflows of github to reduce amount of time of formatting code manually and dealing with human errors.
- Modifying  Pipfile to consist of black settings to support running locally the command of black for formatting check: 
1. vagrant ssh 
2. cd /vagrant
3. pipenv run python -m black . --check --skip-string-normalization
4. (For formatting the files repeat 1 to 3 and eliminate the --check argument)

closes #78  
closes #79

BE AWARE: Apparently, Black formatting prefer double quote than single quote, couldn't find a simple way to make it ignore single quotes, so the formatting files are mostly as a result of this.

### CODE REVIEW

File: .github/workflows/black.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      - name: Install Pipenv and Git
+        run: dnf install -y pipenv git 
+      - name: Checkout code
+        uses: actions/checkout@v2
+        name: Setup environment
+        run: pipenv sync --dev 
+        name: Run Black
+        uses: psf/black@stable
+        with: 
+        args: ". --check"
```
Comment (EdDev) [inline .github/workflows/black.yml]: If you want to reduce the string single or double quote changes, you can add `--skip-string-normalization`.

File: .github/workflows/black.yml [hunk 2]
```diff
@@ -4,16 +4,16 @@ on: [push, pull_request]
 
 jobs: 
   Black:
-    runs-on: ubuntu-20.4
+    runs-on: ubuntu-latest
```
Comment (EdDev) [inline .github/workflows/black.yml:7]: You should move this changes to the prev commit (it appears on the 2nd).

File: .github/workflows/black.yml (hunk 1 above)
Comment (maayanhd) [reply .github/workflows/black.yml]: am I adding this in the 'args' section? And should I do that? what's the best practice?

Comment (EdDev) [reply .github/workflows/black.yml]: It is an argument of `black`, so I do not understand your question.

Regarding if this option is appropriate or not is debatable and is context dependent.
Personally, I like single quotes because I find it cleaner (less pixels), so in the projects I was involved I did used this option. On the other hand, it makes it harder to enforce a format, as you will need to comment about it in reviews to keep it in this standard.
In this case, you can even do it in two phases: Do not enforce it at start, and in a following change do enforce it. But as the codebase is not that big, I think you do not need this steps, just decide what you prefer.

Bottom line, it is up to you. Just reason for your action and decision in the commit so it will be recorded for the future.

Comment (maayanhd) [reply .github/workflows/black.yml]: I was just making sure since I'm still not familiar entirely with YML files. Regarding my decision, I definitely prefer single quotes in most places it already has been, also as it was a default in some of Django default files, I see this formatting files commit as too vast and unnecessary at this point, I will make the changes and commit the formatted files after adding this argument, and later we can decide publicly where we do prefer double quotes if any in our code without using the formatting of Black.

**Comment by maayanhd** [comment]
@EdDev For some reason the argument of skipping the string normalization is working locally but not on the CI, do you know what the cause of that? 
image

as appears on lines 13-14 it's not ignoring the formatting from single to double quotes-
image

File: .github/workflows/black.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      - name: Install Pipenv and Git
+        run: dnf install -y pipenv git 
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Setup environment
+        run: pipenv sync --dev 
+      - name: Run Black
+        uses: psf/black@stable
+        with: 
+          args: ". --check --skip-string-normalization"
```
Comment (EdDev) [inline .github/workflows/black.yml:19]: I think it will be better to run it with the Pipfile content:
```
run: pipenv run python -m black . --check --skip-string-normalization
```
Maybe even try the direct one: `pipenv run black . --check --skip-string-normalization`

It will be in sync with what you expect to run locally.

**Review by EdDev — COMMENTED** [review]
Try the inline suggestion.

File: .github/workflows/black.yml (hunk 3 above)
Comment (maayanhd) [reply .github/workflows/black.yml]: I replaced lines 17-19 with both of your suggestions and for some reason now the CI is not triggered at all after force pushing as it did. It also happened to us with flake8 at the time, but I'm not sure how to solve it, it had something to do with new changes or something but I really can't find a solution for this.

Comment (EdDev) [reply .github/workflows/black.yml]: Isn't it because there are conflicts now?

Comment (maayanhd) [reply .github/workflows/black.yml]: oh ok, didn't notice..

Comment (maayanhd) [reply .github/workflows/black.yml]: @Yarboa @Leeadar it seems the PyTest fails for the changes of the merged PR again on the same error, isn't it was supposed to be fixed already?

Comment (EdDev) [reply .github/workflows/black.yml]: It is most likely your rebase that caused it.
Try to run the tests without your last commit locally. And re-do the formatting (the last commit) from scratch and test.

Comment (maayanhd) [reply .github/workflows/black.yml]: Finally, thanks, great idea @EdDev .

**Review by EdDev — APPROVED** [review]
Very nice, thanks!

**Review by Leeadar — APPROVED** [review]
LGTM

**Review by Yarboa — APPROVED** [review]
